library/aria.c

```diff
@@ -451,7 +451,7 @@ int mbedtls_aria_setkey_enc( mbedtls_aria_context *ctx,
     uint32_t w[4][4], *w2;
 
     if( keybits != 128 && keybits != 192 && keybits != 256 )
-        return( MBEDTLS_ERR_ARIA_INVALID_KEY_LENGTH );
```

Comment:
Isn't returning `MBEDTLS_ERR_ARIA_BAD_INPUT_DATA` where we used to return `MBEDTLS_ERR_ARIA_INVALID_KEY_LENGTH` an API break? I see `MBEDTLS_ERR_ARIA_INVALID_KEY_LENGTH` is not deprecated, so do we need this code change?

Reply:
It looks there there may be similar issues with Blowfish and Camelia as well.

Reply:
@Patater `MBEDTLS_ERR_INVALID_KEY_LENGTH` is deprecated and replaced by `MBEDTLS_ERR_BAD_INPUT_DATA` sharing the same error code, so the change (a) is necessary if we require not to use deprecated API anymore, and (b) doesn't change observable behaviour.

Reply:
This change is OK.

Sorry, I misread the constant name. KEY_LENGTH vs INPUT_LENGTH: two different things.

Reply:
Yes, unfortunately so - that's the point.